fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -515,9 +514,20 @@ protected void addExternalContentHeaders(final FedoraResource resource) {
     }
 
     private void addAclHeader(final FedoraResource resource) {
-        if (!(resource instanceof FedoraWebacAcl) && !resource.isMemento()) {
-            final String resourceUri = getUri(resource.getDescribedResource()).toString();
-            final String aclLocation =  resourceUri + (resourceUri.endsWith("/") ? "" : "/") + FCR_ACL;
+        if (!resource.isAcl()) {
+
+            final FedoraResource aclOwner;
+
+            if (resource.isTimeMap() || resource.isMemento()) {
+                final FedoraResource aclOwnerCandidate = resource.getOriginalResource();
+                aclOwner =
+                    aclOwnerCandidate.isDescription() ? aclOwnerCandidate.getDescribedResource() : aclOwnerCandidate;
```

Comment:
You should be able to just do 
```
aclOwner = resource.getOriginalResource().getDescribedResource();
```
`getDescribedResource()` will return itself if called on the described resource.

Reply:
modified.